### PR TITLE
add reset password function

### PR DESCRIPTION
**Pull request by jetmirsyla**
**Summary**
Admins should be able to reset password
[Asana Ticket](https://app.asana.com/0/1145262842098022/1206133849022257)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -78,6 +80,8 @@ class GoogleAuth extends DbBase {
         blockPrivilege TINYINTEGER,
         analyticsPrivilege TINYINTEGER,
         manageAdminsPrivilege TINYINTEGER,
+        passwordResetToken TEXT,
```
Comment (grantfayvor) [inline index.js:83]: add a migration for these fields in the migration file - https://github.com/bitfinexcom/bfx-facs-auth-google/blob/master/migrations.js

File: index.js [hunk 2]
```diff
@@ -468,6 +472,33 @@ class GoogleAuth extends DbBase {
...
+  async resetAdminPassword(email, newPassword, passwordResetToken) {
+    assert.ok(this.conf.useDB, 'Cannot add admins if DB is not available')
+
+    assert.ok(typeof email === 'string', 'Email is required')
+    assert.ok(typeof newPassword === 'string', 'New Password is required')
+
+    const admin = await this._getAdmin(email)
+    if (!admin) throw new UserError('ADMIN_ACCOUNT_DOES_NOT_EXIST_OR_IS_NOT_ACTIVE')
+    if (admin.passwordResetToken !== passwordResetToken) throw new UserError('INVALID_passwordResetToken')
+    let expiryDate = new Date(admin.passwordResetSentAt)
```
Comment (grantfayvor) [inline index.js:484]: should be fine to use `const` here

```suggestion
    const expiryDate = new Date(admin.passwordResetSentAt)
```

File: index.js [hunk 3]
```diff
@@ -550,7 +581,7 @@ class GoogleAuth extends DbBase {
   async getAdmin (email) {
     const admin = await this._getAdmin(email)
     const displayKeys = ['email', 'level', 'blockPrivilege', 'company',
-      'analyticsPrivilege', 'manageAdminsPrivilege', 'readOnly', 'active', 'timestamp', FORMS_FIELD]
+      'analyticsPrivilege', 'manageAdminsPrivilege', 'readOnly', 'active', 'timestamp', 'passwordResetToken', 'passwordResetSentAt', FORMS_FIELD]
```
Comment (grantfayvor) [inline index.js:584]: no reason to return these fields when fetching admins. they are only useful in the reset password process.

**Review by grantfayvor — CHANGES_REQUESTED** [review]
some comments

**Review by grantfayvor — APPROVED** [review]
lgtm!

**Review by ezewer — APPROVED** [review]
looks good to me!

**Pull request merged**